browser/src/Editor/NeovimEditor/NeovimEditor.tsx

```diff
@@ -741,25 +741,20 @@ export class NeovimEditor extends Editor implements IEditor {
             this._neovimInstance.command(`:e! ${path}`)
         })
 
-        // TODO: Factor this out to a react component
-        // enable opening a file via drag-drop
-        document.body.ondragover = ev => {
+        // This is necessary to prevent electron's default behaviour on drag and dropping
+        // which replaces the webContent aka the entire editor with the text, NOT Good
+        // also DO Not Stop Propagation as this breaks other drag drop functionality
+        document.addEventListener("dragover", ev => {
```

Comment:
I was hoping with this change we'd be able to remove this dependency - it's not ideal for `NeovimEditor` to have this global side effect (for example, if we have multiple `NeovimEditor`s). Could we factor this behavior into the `FileDropHandler`?

Reply:
@bryphe an alternative option option is to call `stopPropagation` but depending on if we are very clear about how future drag and drop functionality will work it might be fine otherwise I think this could cause unusual long term bugs e.g. if there is another drag or drop bit of functionality that perhaps overlaps with this one then another contributor who is not aware of the inner working of the fileDropHandler wouldn't understand why their drop event was not being dispatched not knowing that somewhere along the line the event's propagation was halted.

I haven't tried it but if we `stopPropagation` in the target element in this case the editor container, it wont move outside to the document. This will mean though that if down the line there is the desire for some functionality maybe drag something into some specific section of the editor for some other reason that won't work

All we really want is to prevent the *document's* global behaviour but otherwise have events bubble up as normal was my thinking hence the global listeners